chart-templates/crd-assume-ownership/templates/jobs.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Chart.Name }}-create
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade, post-rollback
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: {{ .Chart.Name }}-create
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      serviceAccountName: {{ .Chart.Name }}-manager
+      containers:
+        - name: kubectl
+          image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
+          imagePullPolicy: IfNotPresent
+          command:
+          - /bin/kubectl
+          - apply
+          - -f
+          - /etc/config/crd-manifest.yaml
+          volumeMounts:
+            - name: crd-manifest
+              readOnly: true
+              mountPath: /etc/config
+      restartPolicy: OnFailure
+      volumes:
+      - name: crd-manifest
+        configMap:
+          name: {{ .Chart.Name }}-manifest
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Chart.Name }}-delete
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: {{ .Chart.Name }}-delete
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      serviceAccountName: {{ .Chart.Name }}-manager
+      containers:
+        - name: kubectl
+          image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
+          imagePullPolicy: IfNotPresent
+          command:
+          - /bin/kubectl
+          - delete
+          - -f
+          - /etc/config/crd-manifest.yaml
```

Comment:
We may need to have more advanced script in here long term if finalizer blocks the delete. https://github.com/kubernetes/kubernetes/issues/60538#issuecomment-369099998

Reply:
Hmm this one is a little tricky. If we want to do the patch based logic that is described in the comment mentioned above, we'll need to probably switch this PR to using the `rancher/hyperkube` image instead the `rancher/kubectl` image, since the `rancher/kubectl` image does not come with `/bin/sh` installed iirc (preventing us from running multiple `kubectl` commands unless we use one initContainer per CRD to uninstall).

Also, in order to get the CRD names to apply the finalizer patches, we would have to continue to use the concept of setting CRDs to env vars in the script (flagged in https://github.com/rancher/charts/pull/595#discussion_r483970192 as something we want to avoid).

As of now, I'm going to just add an initContainer that does a `kubectl patch -f /etc/config/crd-manifest.yaml` before running `kubectl delete -f /etc/config/crd-manifest`, which will effectively remove any declared finalizer blocks by "resetting" the CRD before deleting it; hopefully, this should resolve any issues brought up by https://github.com/kubernetes/kubernetes/issues/60538#issuecomment-369099998.

Reply:
Yea I'm not sure around likelihood of this actually happening with our charts.